NorthstarDLL/httprequesthandler.cpp

```diff
@@ -0,0 +1,504 @@
+template<ScriptContext context>
+SQRESULT SQ_InternalMakeHttpRequest(HSquirrelVM* sqvm)
+{
+	if (!g_httpRequestHandler || !g_httpRequestHandler->IsRunning())
+	{
+		spdlog::warn("NS_InternalMakeHttpRequest called while the http request handler isn't running.");
+		g_pSquirrel<context>->pushinteger(sqvm, -1);
+		return SQRESULT_NOTNULL;
+	}
+
+	if (Tier0::CommandLine()->FindParm("-disablehttprequests"))
+	{
+		spdlog::warn("NS_InternalMakeHttpRequest called while the game is running with -disablehttprequests."
+					 " Please check if requests are allowed using NSIsHttpEnabled() first.");
+		g_pSquirrel<context>->pushinteger(sqvm, -1);
+		return SQRESULT_NOTNULL;
+	}
+
+	HttpRequest request;
+	request.method = static_cast<HttpRequestMethod::Type>(g_pSquirrel<context>->getinteger(sqvm, 1));
+	request.baseUrl = g_pSquirrel<context>->getstring(sqvm, 2);
+
+	SQTable* headerTable = sqvm->_stackOfCurrentFunction[3]._VAL.asTable;
+	for (int idx = 0; idx < headerTable->_numOfNodes; ++idx)
+	{
+		tableNode* node = &headerTable->_nodes[idx];
+
+		// TODO: Figure out why the first three nodes of this table are OT_NULL with a single value.
+		// Then re-enable this if possible.
+		//if (node->key._Type != OT_STRING || node->val._Type != OT_STRING)
+		//{
+		//	g_pSquirrel<context>->raiseerror(
+		//		sqvm, fmt::format("Invalid header type or value (expected string, got {} = {})", SQTypeNameFromID(node->key._Type), SQTypeNameFromID(node->val._Type)).c_str());
+		//	return SQRESULT_ERROR;
+		//}
+
+		if (node->key._Type == OT_STRING && node->val._Type == OT_STRING)
+		{
+			request.headers[node->key._VAL.asString->_val] = node->val._VAL.asString->_val;
+		}
+	}
+
+	SQTable* queryTable = sqvm->_stackOfCurrentFunction[4]._VAL.asTable;
+	for (int idx = 0; idx < queryTable->_numOfNodes; ++idx)
+	{
+		tableNode* node = &queryTable->_nodes[idx];
+		// Same as above.
+		//if (node->key._Type != OT_STRING || node->val._Type != OT_STRING)
+		//{
+		//	g_pSquirrel<context>->raiseerror(
+		//		sqvm, fmt::format("Invalid query parameter type or value (expected string, got {} = {})", SQTypeNameFromID(node->key._Type), SQTypeNameFromID(node->val._Type)).c_str());
+		//	return SQRESULT_ERROR;
+		//}
+
+		if (node->key._Type == OT_STRING && node->val._Type == OT_STRING)
+		{
+			request.queryParameters[node->key._VAL.asString->_val] = node->val._VAL.asString->_val;
+		}
+	}
+
+	request.contentType = g_pSquirrel<context>->getstring(sqvm, 5);
+	request.body = g_pSquirrel<context>->getstring(sqvm, 6);
+	request.timeout = g_pSquirrel<context>->getinteger(sqvm, 7);
+	request.userAgent = g_pSquirrel<context>->getstring(sqvm, 8);
+
+	int handle = g_httpRequestHandler->MakeHttpRequest<context>(request);
+	g_pSquirrel<context>->pushinteger(sqvm, handle);
+	return SQRESULT_NOTNULL;
+}
+
+// bool NSIsHttpEnabled()
+template<ScriptContext context> SQRESULT SQ_IsHttpEnabled(HSquirrelVM* sqvm)
+{
+	const bool bIsDisabled = Tier0::CommandLine()->FindParm("-disablehttprequests");
```

Comment:
This could be static.

Reply:
Resolved on new branch, will re-make PR.